### PR TITLE
改成相对路径，加上配置文件 Fix #13 Fix #9 Fix#11

### LINKED ISSUE #9: 借车信息填写改进
希望把性别那一栏做成默认为空的必填项，已经有新会员忘了改性别了，结果报名是男，借车是女……

### LINKED ISSUE #13: config.php
Use config.php instead of raw username/password for database connection.

### PR DESCRIPTION
**Pull request by zhzhzoo**

### CODE REVIEW

File: api/client.php [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?php
+    require_once '../config.php';
```
Comment (huxuan) [inline api/client.php]: I don't like RELATIVE path especially for those are not in the same directory.
The more we use `../`, the code base will be harder to maintain.
I prefer to use `require  $_SERVER["DOCUMENT_ROOT"].'/config.php'`
If you have more elegent suggestions, feel free to challenge me. :-)

File: api/client.php [hunk 2]
```diff
@@ -290,7 +291,7 @@ function register() {
 
 	function seemain() {
 		echo '<capu><info><code>-1</code>';
-		$con=mysql_connect("localhost","root","19951025");
+		@$con=mysql_connect(CAPU_DB_HOST, CAPU_DB_USER, CAPU_DB_PWD);
```
Comment (huxuan) [inline api/client.php]: Yeah, I really appreciate this change. That's just what I want to do when I propose issue #13.
But there is also another concern, whether we can only use ONE `$con` instance all over the code base?
And for `mysql_connect`, it is deprecated, let's change it in the future.
We can use `MySQLi` extension [1] with its contruction method [2].
[1] http://php.net/manual/en/class.mysqli.php
[2] http://php.net/manual/en/mysqli.construct.php

File: api/jiekouapi.php [hunk 3]
```diff
@@ -428,7 +429,6 @@ function updatetokentime($con,$token,$ip) {
 	
 	function post($con,$token,$bid,$ip,$attachs) {
 		$time=time();
-		echo '<capu>';
```
Comment (huxuan) [inline api/jiekouapi.php]: Thanks for fixing this small bug. Just one small suggestion, you can send a different PR for this. As I have explained in other comments. It's not a good idea to bring too many (not really related) modifications in one PR.

File: assets/css/base.css [hunk 4]
```diff
@@ -10,7 +10,7 @@ html {
 }
 body{
     font: 14px/24px "Microsoft YaHei","Arial",sans-serif;
-    background:url(/assets/images/beige_paper.png) left top repeat;
+    background:url(../../assets/images/beige_paper.png) left top repeat;
```
Comment (huxuan) [inline assets/css/base.css]: Can you explain more about all these changes?
I agree with you that relative path in PHP is a little buggy.
But this is HTTP and it handles paths from root directory really well.
I don't think we need these changes.

File: api/client.php [hunk 5]
```diff
@@ -25,7 +26,7 @@
 	function request($posts) {
 		$token=@$_POST['token'];
 		$ip=$_SERVER['REMOTE_ADDR'];
-		$url="http://127.0.0.1/api/jiekouapi.php?ip=$ip&token=$token";
+		$url=CAPU_JIEKOUAPI . "?ip=$ip&token=$token";
```
Comment (huxuan) [inline api/client.php]: I really like this style!

File: bbs/main/index.php [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <?php
-	include("../lib/mainfunc.php");
+    //$cl = microtime(true);
```
Comment (huxuan) [inline bbs/main/index.php]: Hummm... It's not a good idea to hide tiny candy (like useless comments). ;-)

File: bbs/main/index.php [hunk 7]
```diff
@@ -393,6 +394,7 @@ function packjump($bid,$p,$text,$extr){
 	 </div>
 
 <div class="footer">
+<!--<p>执行用<?php echo(number_format((microtime(true)-$cl)*1000, 3));?>ms</p>--!>
```
Comment (huxuan) [inline bbs/main/index.php]: Please remember to revert the changes for debugging.

File: config-sample.php [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (huxuan) [inline config-sample.php]: I prefer to add something like following comments at the beginning part. Feel free to revise as you like.
```
/**
 * Basic configuration of CAPUBBS.
 *
 * This file contains the following configuration: MySQL settings
 *
 * You need to copy this file and rename it to 'config.php' to make it work.
 * You also need to change the corresponding setting to its actual value. For
 * more information, please look at the comments before each item.
 */
```

File: bbs/lib/mainfunc.php [hunk 9]
```diff
@@ -1,11 +1,11 @@
 <?php
+require_once "../../config.php";
 function mainfunc($posts,$debug=false){
 	$ip=$_SERVER["REMOTE_ADDR"];
 	@$token=$_COOKIE['token'];
-	$url="http://127.0.0.1/api/jiekouapi.php?ip=$ip&token=$token";
+	$url=CAPU_JIEKOUAPI."?ip=$ip&token=$token";
```
Comment (huxuan) [inline bbs/lib/mainfunc.php]: We need to set a rule for using `A . B` or `A.B`. The difference is whether we have space before and after the period. But there is another rule we need to follow: we should only use one style all over the code base.

File: bbs/register/action.php [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <?php
-include("../lib/mainfunc.php");
-include("../../assets/api/captcha.php");
+require_once "../lib/mainfunc.php";
+require_once "../../assets/api/captcha.php";
```
Comment (huxuan) [inline bbs/register/action.php]: Can you explain more detail about whether you use `require_once` or just `require`?

**Comment by huxuan** [comment]
Another suggestion is that you can add `Fix #13` in the title of Pull Request. With that the issue can be close automatically after your pull request being merged. This github feature is really useful for issues management. For more information about this, feel free to look over here[1]. 

[1] https://guides.github.com/features/issues/

File: api/client.php (hunk 1 above)
Comment (zhzhzoo) [inline api/client.php]: 嗯主要是我的服务器配置比较奇葩，CAPUBBS 没有当成 url 和网站文件的根，然后就不得不这么做了。。。

File: api/client.php (hunk 2 above)
Comment (zhzhzoo) [inline api/client.php]: 关于一个 `$con` 嗯re。。。都手写了一遍改得全部改掉，是不是提出来写在一起比较好。。。

然后那个 deprecated，好像改起来得把查询的代码也改掉好麻烦得以后了。。。

File: api/jiekouapi.php (hunk 3 above)
Comment (zhzhzoo) [inline api/jiekouapi.php]: 嗯好的~ 知道啦以后发另外一个

File: assets/css/base.css (hunk 4 above)
Comment (zhzhzoo) [inline assets/css/base.css]: 嗯这个也是我把论坛放在 /bbs/xxx/ 下面的问题。。。。

File: bbs/main/index.php (hunk 6 above)
Comment (zhzhzoo) [inline bbs/main/index.php]: 据说论坛打开很慢我就想给生产环境也加一个计时器。。。但是现在还没想好就先注释掉了。。。

File: bbs/lib/mainfunc.php (hunk 9 above)
Comment (zhzhzoo) [inline bbs/lib/mainfunc.php]: 嗯以及所有二元运算符和逗号~~

Comment (zhzhzoo) [inline bbs/lib/mainfunc.php]: 还有缩进几个空格，第一行代码缩不缩进~

File: bbs/register/action.php (hunk 10 above)
Comment (zhzhzoo) [inline bbs/register/action.php]: 嗯 `require_once` 发现 require 过了的话就不 require 了，`require` 的话发现 require 过了还是会 require ~~

File: config-sample.php (hunk 8 above)
Comment (zhzhzoo) [inline config-sample.php]: 好主意！

File: api/client.php (hunk 1 above)
Comment (huxuan) [inline api/client.php]: I would like to argue more for this code style. This kind of relative path import is not good for further code maintenance. We can change the Server configuration. We can also set up some kind of contant like `CAPU_ROOT` to the root directory for us.

File: api/client.php (hunk 5 above)
Comment (huxuan) [inline api/client.php]: Let's set up the first code convention. [1] ;-)

Please change all other similar changes to fits this convention.

https://github.com/CAPU-ENG/CAPUBBS/wiki/Convention#code-style

File: api/client.php (hunk 2 above)
Comment (huxuan) [inline api/client.php]: OK, I agree with you to change that in another PR.

File: assets/css/base.css (hunk 4 above)
Comment (huxuan) [inline assets/css/base.css]: We need to talk about this. I think there must be some misunderstanding between us.

File: bbs/lib/mainfunc.php (hunk 9 above)
Comment (huxuan) [inline bbs/lib/mainfunc.php]: I have add some into Conventions.
Please modify all related changes made by this PR.

File: bbs/main/index.php (hunk 6 above)
Comment (huxuan) [inline bbs/main/index.php]: It's not a good idea to add these comments in this PR. Please REMOVE it before it is ready for actual production use.

File: api/client.php (hunk 5 above)
Comment (zhzhzoo) [inline api/client.php]: 我觉得我们应该禁止所有隐式类型转换。。。
